src/http/mod.rs

```diff
@@ -149,6 +153,24 @@ impl ResponseBody {
         self.inner.into_inner()
     }
 }
+impl ResponseBody {
+    pub async fn bytes(&mut self) -> Result<Vec<u8>, io::Error> {
```

Comment:
Let's add `limit: Option<usize>` to this and the other two methods using `take()`.
Implementation can be mostly copied from `body_vec` a few lines above.

All methods in ResponseExt can be removed.